analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Sonar/SonarExplodedGraph.cs

```diff
@@ -1004,6 +1004,12 @@ private ProgramState VisitMemberAccess(MemberAccessExpressionSyntax memberAccess
                         newProgramState = newProgramState.StoreSymbolicValue(symbol, sv);
                     }
                 }
+                else if (symbol is IMethodSymbol { IsExtensionMethod: true } methodSymbol
+                    && methodSymbol.GetFirstSyntaxRef() is MethodDeclarationSyntax syntax
+                    && syntax.ParameterList.Parameters.FirstOrDefault().AttributeLists.SelectMany(list => list.Attributes).Any(x => x.ToStringContains("ValidatedNotNull")))
```

Comment:
I'm not sure I have a better option than the `ToString`, unless I query the symbol for the attributes to verify the type, like here: https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Sonar/InvocationVisitor.cs#L243

I guess in practice there won't be that many extension methods with lots of attributes on the first parameter so it may be cleaner

Reply:
I agree with this approach, we only look at attributes on extension methods, and there should not be that many in practice, just like you said. And this way we don`t have to obtain the symbol for the parameter.

Reply:
I've just realized that `FirstOrDefault` is used there without a null check. 
In any case an extension method should always have the first `this param`, however then it should be `First()` instead of `FirstOrDefault()`

I've opened https://github.com/SonarSource/sonar-dotnet/pull/5233 for a defensive check